Ajout de nouvelles stratégies de pilotage dans la description

josephmure opened the pull request:
- Nouvelles stratégies non financées (S_s non défini)
- Suppression de l'explication des stratégies spécifiant R_{NVs}
- Explications supplémentaires quant aux différentes stratégies
- Le code des nouvelles stratégies reste à écrire ! Plus tard...

josephmure committed:
Explicite la signification des variables intermédiaires

mbaudin47 committed:
Merge pull request #2 from josephmure/description

Explicite la signification des variables intermédiaires

josephmure committed:
Ajout de nouvelles stratégies de pilotage dans la description
- Nouvelles stratégies non financées (S_s non défini)
- Suppression de l'explication des stratégies spécifiant R_{NVs}
- Explications supplémentaires quant aux différentes stratégies
- Le code des nouvelles stratégies reste à écrire ! Plus tard...

mbaudin47 commented on doc/Description-du-composant-retraites.ipynb:
```diff
@@ -82,35 +82,86 @@
+    "\n",
+    "Les variables $T_s$, $P_s$, $A_s$ et $S_s$ sont liées par une équation. On appelle *pilotage* le choix des 3 variables définies par l'utilisateur. Dans cette librairie, nous proposons d'inverser les calculs du COR pour permettre à l'utilisateur de choisir n'importe quel pilotage parmi les 4 possibles :\n",
+    "\n",
+    "| Entrées | Sortie | Commentaire |\n",
```
Est-ce que tu pourrais s'il te plaît ajouter une colonne avec l'indice du pilotage ? En effet, numéroter les pilotages permet de les retrouver sans ambiguïté dans la suite.

josephmure replied:
Fait

mbaudin47 commented on doc/Description-du-composant-retraites.ipynb:
```diff
@@ -82,35 +82,86 @@
+    "\n",
+    "Le système d'équations ci-dessus (définition de $D_s$ et modèle du COR) est un système de 2 équations à 5 inconnues ($S_s$, $P_s$, $T_s$, $A_s$, $D_s$). Piloter le système consiste à fixer 3 de ces 5 inconnues. Les 4 pilotages évoqués ci-dessus restent possibles, bien entendu :\n",
+    "\n",
+    "| Entrées | Sorties | Commentaire |\n",
```
De même, est-ce que tu peux ajouter une colonne avec les indices des pilotages ?

josephmure replied:
Fait

mbaudin47 commented on doc/Description-du-composant-retraites.ipynb:
```diff
@@ -82,35 +82,86 @@
+    "\n",
+    "Mais on peut aussi en considérer 4 autres (et seulement 4 autres) :\n",
+    "\n",
+    "| Entrées | Sorties | Commentaire |\n",
```
Idem.

mbaudin47 reviewed (changes requested):
Super ! Il n'y a que quelques détails à modifer à mon avis.

josephmure committed:
Changements demandés dans le fichier description
- numéros des pilotages ajoutés aux tableaux
- ordre des paramètres changé : S, P, A, T, D utilisé systématiquement

josephmure committed:
Code des nouveaux pilotages
7 nouvelles méthodes SimulateurRetraites.pilotageParXxxYyyZzz où
Xxx, Yyy, Zzz sont les 3 paramètres pilotés

josephmure reviewed (commented):
Les numéros des pilotages ont été ajoutés aux tableaux.
Peux-tu s'il te plaît vérifier le code des nouveaux pilotages ?
Je n'ai retiré aucun pilotage, même si ceux qui font intervenir R_{NVs} ne sont plus mentionnés dans le fichier Description.

mbaudin47 commented:
Je ne suis pas très favorable pour retirer les équations qui indiquent comment fonctionnent les pilotages fondés sur RNV. On est d'accord pour dire que cette variable devrait être écartée dans nos analyse. Toutefois, 
* C'est une variable du modèle du COR (que nous n'avons malheureusement pas choisie, mais c'est ainsi). 
* Le code existe déjà et fonctionne de telle sorte que si nous effacons ces équations, le code correspondant n'aura plus de documentation mathématique. 
* Bruno a déjà rédigé du matériel sur la base de cette variable (par exemple, la page principale du README ainsi que certains articles sur Médiapart). En conséquence, si nous supprimons la documentation de ces équations, les articles ne pourront plus s'appuyer que sur le code, avec moins de facilité de faire le lien entre les résutlats et le code. Je pense en particulier à des nouveaux contributeurs qui souhaiteraient vérifier le code, les équations et les résultats. 

Si les équations des pilotages associées à RNV sont en annexe du document (à la fin du document), est-ce que cela te conviendrait : la présentation principale serait inchangée, mais le matériel serait toujours accessible. Ca te conviendrait ?

mbaudin47 commented on doc/Description-du-composant-retraites.ipynb:
```diff
@@ -132,76 +183,77 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Pilotage 1 : calcul  à âge et niveau de vie défini\n",
```
Il est dommage de perdre ce matériel (voir mon commentaire).

josephmure replied:
Ce matériel est réintégré avec une présentation homogène à celle des autres pilotages.

mbaudin47 commented on retraites/SimulateurRetraites.py:
```diff
@@ -59,14 +59,238 @@ def __init__(self, json_filename):
 
     def pilotageCOR(self):
         """
-        pilotage 0 : statu quo du COR
```
Oui, démarrer à 1 est plus logique. C'est OK.

mbaudin47 reviewed (changes requested):
Le code me semble OK. 

Il manque les tests unitaires. Ils sont dans le dossier tests. Pour chaque pilotage, un test unitaire vérifie que les variables imposées sont égales à la contrainte et que les autres variables ont les valeurs requises avec une tolérance plus ou moins exigeante : 

https://github.com/brunoscherrer/retraites/blob/master/tests/test_retraites.py

Faire les tests unitaires demande un peu de temps, mais c'est efficace et assez facile dans ce cas particulier.

josephmure committed:
Merge branch 'CorrectionDescription' into description

josephmure commented:
Le conflit est résolu. A présent, je vais travailler à la réintégration des explications concernant les pilotages faisant intervenir R_{NVs}.

josephmure committed:
Les 2 pilotages R_NVs réintégrés à la Description
Leur présentation est rendue homogène à celle des autres pilotages
mais ils sont présentés dans une section séparée et
les raisons de ne pas utiliser R_NVs sont mieux expliquées

josephmure commented:
> Si les équations des pilotages associées à RNV sont en annexe du document (à la fin du document), est-ce que cela te conviendrait : la présentation principale serait inchangée, mais le matériel serait toujours accessible. Ca te conviendrait ?

Plutôt qu'en annexe, je les ai mis à la suite dans une rubrique "Exemples d'autres pilotages". Il n'y en a que 2, en fait. Je les ai introduits en disant que le fait de considérer en plus la variable R_NVs ouvre la possibilité d'autres pilotages et que la liste qui en est donnée n'est pas exhaustive.

josephmure committed:
Plusieurs améliorations concernant les pilotages :
- ajout du pilotageParPensionAgeCotisations (celui du COR avec valeurs
modifiables)
- MàJ docstring __init__ de la classe SimulateurRetraites

mbaudin47 reviewed (approved):
C'est beau !

josephmure commented:
Attention ! je n'ai pas encore implémenté les tests !

mbaudin47 commented:
Ok, j'attends les tests unitaires avant de créer la PR vers Bruno.
